src/lex.cpp

```diff
@@ -1288,6 +1292,52 @@ void lexer::parse_modern_octal_number() {
       diag_unexpected_characters_in_octal_number>(input);
 }
 
+void lexer::check_precision_loss(const char8* number_begin,
```

Comment:
This function only works if the number has no decimal or exponent. Make this clearer by mentioning "integer" in the function name? Or add a doc comment?